test/conformance/errorcondition_test.go

```diff
@@ -118,79 +118,97 @@ func TestContainerErrorMsg(t *testing.T) {
 func TestContainerExitingMsg(t *testing.T) {
 	t.Parallel()
-	clients := setup(t)
-
-	names := test.ResourceNames{
-		Config: test.ObjectNameForTest(t),
-		Image:  "failing",
-	}
-
 	const (
 		// The given image will always exit with an exit code of 5
 		exitCodeReason = "ExitCode5"
 		// ... and will print "Crashed..." before it exits
 		errorLog = "Crashed..."
 	)
 
-	t.Logf("Creating a new Configuration %s", names.Image)
-
-	// This probe is crucial for having a race free conformance test. It will prevent the
-	// pod from becoming ready intermittently.
-	probe := &corev1.Probe{
-		Handler: corev1.Handler{
-			HTTPGet: &corev1.HTTPGetAction{},
+	tests := []struct {
+		Name           string
+		ReadinessProbe *corev1.Probe
+	}{{
+		Name: "http",
+		ReadinessProbe: &corev1.Probe{
+			Handler: corev1.Handler{
+				HTTPGet: &corev1.HTTPGetAction{},
+			},
 		},
-	}
-	if _, err := test.CreateConfiguration(t, clients, names, &test.Options{ReadinessProbe: probe}); err != nil {
-		t.Fatalf("Failed to create configuration %s: %v", names.Config, err)
-	}
-	defer test.TearDown(clients, names)
-	test.CleanupOnInterrupt(func() { test.TearDown(clients, names) })
+	}, {
+		Name: "tcp",
+		ReadinessProbe: &corev1.Probe{
+			Handler: corev1.Handler{
+				TCPSocket: &corev1.TCPSocketAction{},
+			},
+		},
+	}}
 
-	t.Log("When the containers keep crashing, the Configuration should have error status.")
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.Name, func(t *testing.T) {
+			t.Parallel()
+			clients := setup(t)
 
-	err := test.WaitForConfigurationState(clients.ServingClient, names.Config, func(r *v1alpha1.Configuration) (bool, error) {
-		cond := r.Status.GetCondition(v1alpha1.ConfigurationConditionReady)
-		if cond != nil && !cond.IsUnknown() {
-			if strings.Contains(cond.Message, errorLog) && cond.IsFalse() {
-				return true, nil
+			names := test.ResourceNames{
+				Config: test.ObjectNameForTest(t),
+				Image:  failing,
 			}
-			t.Logf("Reason: %s ; Message: %s ; Status: %s", cond.Reason, cond.Message, cond.Status)
-			return true, fmt.Errorf("The configuration %s was not marked with expected error condition (Reason=\"%s\", Message=\"%s\", Status=\"%s\"), but with (Reason=\"%s\", Message=\"%s\", Status=\"%s\")",
-				names.Config, containerMissing, errorLog, "False", cond.Reason, cond.Message, cond.Status)
-		}
-		return false, nil
-	}, "ConfigContainersCrashing")
 
-	if err != nil {
-		t.Fatalf("Failed to validate configuration state: %s", err)
-	}
+			t.Logf("Creating a new Configuration %s", names.Image)
 
-	revisionName, err := getRevisionFromConfiguration(clients, names.Config)
-	if err != nil {
-		t.Fatalf("Failed to get revision from configuration %s: %v", names.Config, err)
-	}
+			if _, err := test.CreateConfiguration(t, clients, names, &test.Options{ReadinessProbe: tt.ReadinessProbe}); err != nil {
+				t.Fatalf("Failed to create configuration %s: %v", names.Config, err)
+			}
+			defer test.TearDown(clients, names)
```

Comment:
Move this and the line below up above the CreateConfiguration otherwise this can leak.

Reply:
Thanks for pointing this out. Moved it up to immediately after clients, names creation.